modules/tile-converter/src/i3s-converter/helpers/geometry-converter.ts

```diff
@@ -526,38 +530,77 @@ function convertMesh(
         cartographicOrigin,
         cartesianModelMatrix,
         nodeMatrix: matrix,
-        indices: primitive.indices?.value,
+        indices,
         attributeSpecificTransformation: transformVertexNormals,
         useCartesianPositions: false
       })
     );
     outputAttributes.texCoords = concatenateTypedArrays(
       outputAttributes.texCoords,
-      flattenTexCoords(
-        attributes.TEXCOORD_0 && attributes.TEXCOORD_0.value,
-        primitive.indices?.value
-      )
+      flattenTexCoords(attributes.TEXCOORD_0 && attributes.TEXCOORD_0.value, indices)
     );
 
     outputAttributes.colors = concatenateTypedArrays(
       outputAttributes.colors,
-      flattenColors(attributes.COLOR_0, primitive.indices?.value)
+      flattenColors(attributes.COLOR_0, indices)
     );
 
     if (materialUvRegion) {
       outputAttributes.uvRegions = concatenateTypedArrays(
         outputAttributes.uvRegions,
-        createUvRegion(materialUvRegion, primitive.indices?.value)
+        createUvRegion(materialUvRegion, indices)
       );
     }
 
     outputAttributes.featureIndicesGroups = outputAttributes.featureIndicesGroups || [];
     outputAttributes.featureIndicesGroups.push(
-      flattenBatchIds(getBatchIds(attributes, primitive, images), primitive.indices?.value)
+      flattenBatchIds(getBatchIds(attributes, primitive, images), indices)
     );
   }
 }
 
+type TypedArrayConstructor =
+  | Uint8ArrayConstructor
+  | Int8ArrayConstructor
+  | Uint16ArrayConstructor
+  | Int16ArrayConstructor
+  | Int32ArrayConstructor
+  | Uint32ArrayConstructor
+  | Float32ArrayConstructor
+  | Float64ArrayConstructor;
+
+/**
+ * Converts TRIANGLE-STRIPS to independent TRIANGLES
+ * @param {MeshPrimitive} primitive - the primitive to get the indices from
+ * @returns indices of vertices of the independent triangles
+ */
+function getIndices(primitive: MeshPrimitive): TypedArray {
+  let indices: TypedArray = primitive.indices?.value;
+  if (indices && primitive.mode === GL.TRIANGLE_STRIP) {
+    /*
+    TRIANGLE_STRIP geometry contains n+2 vertices for n triangles;
+    TRIANGLE geometry contains n*3 vertices for n triangles.
+    The conversion from TRIANGLE_STRIP to TRIANGLE implies duplicating adjacent vertices.
+    */
+    const TypedArrayConstructor = indices.constructor as TypedArrayConstructor;
+    const newIndices = new TypedArrayConstructor((indices.length - 2) * 3);
+    let triangleIndex = 0;
+    let currentTriangle = indices.slice(0, 3);
+    newIndices.set(currentTriangle, 0);
```

Comment:
Nit; declare new indices close to its line of use.

Reply:
The type TypedArrayConstructor is not exported in '@math.gl/types'.
There is no "types" in tsconfig.modules.json (math.gl). We will put TypedArrayConstructor to tile-converter/types until it's fixed in math.gl.

Reply:
OK. Looking in the src code it is exported, probably fixed in upcoming 3.7 but not backported to 3.6...